code/modules/modular_computers/file_system/programs/radar.dm

```diff
@@ -0,0 +1,235 @@
+	if(scanning)
+		return
+	objects = list()
+	scanning = TRUE
+	sleep(20) //To limit spamming of the scan button
+	if(program_state == PROGRAM_STATE_KILLED) //If we closed during the scan, cancel the scan
+		scanning = FALSE
+		return
+	for(var/i in GLOB.human_list)
+		var/mob/living/carbon/human/humanoid = i
+		if(!trackable(humanoid))
+			continue
+		var/crewmember_name = "Unknown"
+		if(humanoid.wear_id)
+			var/obj/item/card/id/ID = humanoid.wear_id.GetID()
+			if(ID && ID.registered_name)
+				crewmember_name = ID.registered_name
+		var/list/crewinfo = list(
+			ref = REF(humanoid),
+			name = crewmember_name,
+			)
+		objects += list(crewinfo)
+	scanning = FALSE
+
+/datum/computer_file/program/radar/lifeline/proc/trackable(mob/living/carbon/human/humanoid)
+	if(!humanoid)
+		return FALSE
+	var/turf/here = get_turf(computer)
+	if((humanoid.z == 0 || (humanoid.z == here.z || (is_station_level(humanoid.z) && is_station_level(here.z)))) && istype(humanoid.w_uniform, /obj/item/clothing/under))
+		var/obj/item/clothing/under/uniform = humanoid.w_uniform
+
+		// Suit sensors must be on maximum.
+		if(!uniform.has_sensor || (uniform.sensor_mode < SENSOR_COORDS))
+			return FALSE
+
+		var/turf/there = get_turf(humanoid)
+		return (humanoid.z != 0 || (there && there.z == here.z))
+
+	return FALSE
+
+////////////////////////
+//Nuke Disk Finder App//
+////////////////////////
+
+///A program that tracks crew members via suit sensors
+/datum/computer_file/program/radar/fission360
+	filename = "Fission360"
+	filedesc = "Fission360"
+	program_icon_state = "generic"
+	extended_desc = "This program allows for tracking of nuclear authorization disks and warheads."
+	requires_ntnet = FALSE
+	transfer_access = null
+	available_on_ntnet = FALSE
+	available_on_syndinet = TRUE
+	tgui_id = "NtosRadarSyndicate"
+
+/datum/computer_file/program/radar/fission360/track()
+	var/obj/nuke = locate(selected) in GLOB.poi_list
+	if(!trackable(nuke))
+		return
+
+	var/turf/here_turf = (get_turf(computer))
+	var/turf/target_turf = (get_turf(nuke))
+	var/userot = FALSE
+	var/rot = 0
+	var/pointer="crosshairs"
+	var/locx = (target_turf.x - here_turf.x)
+	var/locy = (here_turf.y - target_turf.y)
+	if(get_dist_euclidian(here_turf, target_turf) > 24) //If they're too far away, we need the angle for the arrow along the edge of the radar display
+		userot = TRUE
+		rot = round(Get_Angle(here_turf, target_turf))
+	else
+		locx = locx + 24
+		locy = locy + 24
+		if(target_turf.z > here_turf.z)
+			pointer="caret-up"
+		else if(target_turf.z < here_turf.z)
+			pointer="caret-down"
+	var/list/trackinfo = list(
+		locx = locx,
+		locy = locy,
+		userot = userot,
+		rot = rot,
+		arrowstyle = "ntosradarpointerS.png",
+		color = "red",
+		pointer = pointer,
+		)
+	return trackinfo
+
+/datum/computer_file/program/radar/fission360/scan()
+	if(scanning)
+		return
+	objects = list()
+	scanning = TRUE
+	sleep(20) //To limit spamming of the scan button
```

Comment:
you could just use a world.time check from last time ran or a callback to set a locking var on the fission360 datum instead of blocking